ui/app/components/totp/key-create.hbs

```diff
@@ -3,39 +3,30 @@
   SPDX-License-Identifier: BUSL-1.1
 ~}}
 
-<form {{on "submit" @create}}>
+<form {{on "submit" (perform @onSubmit)}}>
   <div class="box is-sideless is-fullwidth is-marginless">
     <MessageError @model={{@model}} />
-    <NamespaceReminder @mode={{@mode}} @noun="TOTP key" />
-
-    {{#each (if @model.generate @generatedFields @nonGeneratedFields) as |field|}}
+    {{#each @keyFormFields as |field|}}
       {{#let (find-by "name" field @model.allFields) as |attr|}}
-        {{! TODO add model validations }}
-        <FormField @attr={{attr}} @model={{@model}} data-test-field />
+        <FormField @attr={{attr}} @model={{@model}} @modelValidations={{@modelValidations}} />
       {{/let}}
     {{/each}}
 
-    <Totp::KeyCreateToggleGroups @model={{@model}} />
+    <Totp::KeyCreateToggleGroups @model={{@model}} @modelValidations={{@modelValidations}} @groups={{@groups}} />
   </div>
   <div class="field is-grouped-split box is-fullwidth is-bottomless">
```

Comment:
i think we can remove this div, `<Hds::ButtonSet>` should handle styling for us, though you may need some [spacing](https://github.com/hashicorp/vault/blob/main/ui/app/components/page/userpass-reset-password.hbs#L33)